Clarify that Reserved may also represent private-use in data model (#444)
unicode-org/message-format-wg: spec/data-model/README.md

Implementations MUST NOT rely on the set of `Expression` `body` values being exhaustive,
as future versions of this specification MAY define additional expressions.

> If encountering a `body` with an unrecognised value,
> an implementation SHOULD treat it as it would an `Unsupported` value.

Reviewer: Consider using a more positive statement:

```suggestion
A `body` with an unrecognized value SHOULD be treated as
an `Unsupported` value.
```

Will `body` always have a sigil, that is, will it fit the description below in which unsupported is always reserved or private use? Or do we need to add text below that says (effectively) "reserved, unsupported private-use, or any unrecognized sequence"

}
```

> An `Unsupported` represents an _expression_ with a _reserved_ or _private-use_ _annotation_.

Author: Intentional, because we're at least so far not referring to data model constructs from elsewhere in the spec, and their names would very closely conflict with syntax rules. The data model is currently entirely built on top of the rest, and we should be careful about establishing a dependency on it from elsewhere.

The `sigil` corresponds to the starting sigil of the _annotation_.
The `source` is the "raw" value (i.e. escape sequences are not processed)
and does not include the starting `sigil`.

> **Note**
> Implementers should be aware that future versions of this specification
> might assign meaning to _reserved_ `sigil` values

> > and add new interfaces to this data model.

Author: The fix looks good; it improves on your previous suggestion in https://github.com/unicode-org/message-format-wg/pull/444#discussion_r1278315019.

> > and add new interfaces to this data model.

Reviewer: Wonderful what a cup of coffee ☕ and fresh eyes 👀 will do sometimes...

If the _expression_ includes a _literal_ or _variable_ before the _annotation_,
it is included as the `operand`.

When parsing the syntax of a _message_ that includes a _private-use_ _annotation_
supported by the implementation,

> the implemenation MAY represent it in the data model using a different interface
> as appropriate for the semantics and meaning that it attaches to that _annotation_.

Reviewer: Perhaps a bit stronger:

```suggestion
the implementation SHOULD represent it in the data model
using an interface appropriate for the semantics and meaning
that the implementation attaches to that _annotation_.
```
